### PR TITLE
update instructions for export signing and key ordering

### LINKED ISSUE #1010: signature info ordering
* update .proto spec, signature block on export files, indicating "newest" signature should appear first in the block (both)
    * add limit to number of signatures that is allowed (10)
* update server to actually implement this
* add instructions to admin console. As soon as you make this key active, notify Google/Apple immediately

/kind enhancement

### PR DESCRIPTION
**Pull request by mikehelmick**
Fixes #1010

## Proposed Changes

* Document ordering of signatures on export files when there is more than one
* Document limit (soft) on export signatures
* Update server for this ordering

**Release Note**

```release-note
Export file signatures will be ordered with the most recently created one first. Documented updated expectations around files with multiple signers.
```

### CODE REVIEW

File: internal/export/database/export.go [hunk 1]
```diff
@@ -317,6 +317,8 @@ func (db *ExportDB) LookupSignatureInfos(ctx context.Context, ids []int64, valid
 				SignatureInfo
 			WHERE
 				id = any($1) AND (thru_timestamp is NULL OR thru_timestamp >= $2)
+			ORDER BY
+				id DESC
```
Comment (sethvargo) [inline internal/export/database/export.go:321]: The code comment says time order, but this is id. They are _probably_ the same, but not guaranteed to be.

File: internal/pb/export/export.proto [hunk 2]
```diff
@@ -103,6 +107,11 @@ message TemporaryExposureKey {
 }
 
 message TEKSignatureList {
+  // When their are multiple signatures, they must be sorted in time order
```
Comment (tshortli) [inline internal/pb/export/export.proto:110]:
```suggestion
  // When there are multiple signatures, they must be sorted in time order
```

File: internal/export/database/export_test.go [hunk 3]
```diff
@@ -102,20 +102,25 @@ func TestLookupSignatureInfos(t *testing.T) {
 			SigningKeyID:      "310",
 		},
 	}
-	for _, si := range want {
+	for _, si := range create {
 		if err := New(testDB).AddSignatureInfo(ctx, si); err != nil {
 			t.Fatalf("failed to add signature info %v: %v", si, err)
 		}
 	}
+	// create a 'reverse' order list.
```
Comment (sethvargo) [inline internal/export/database/export_test.go:110]: You can do this without an extra alloc...

```go
for i, j := 0, len(want)-1; i < j; i, j = i+1, j-1 {
  want[i], want[j] = want[j], want[i]
}
```

File: internal/export/database/export_test.go [hunk 4]
```diff
@@ -102,20 +102,25 @@ func TestLookupSignatureInfos(t *testing.T) {
...
 	ids := []int64{want[0].ID, want[1].ID, want[2].ID}
 	got, err := New(testDB).LookupSignatureInfos(ctx, ids, testTime)
 	if err != nil {
 		t.Fatal(err)
 	}
 
-	// The first entry (want[0]) is expired and won't be returned.
-	want = want[1:]
+	// The last entry (want[len-1]) is expired and won't be returned.
+	want = want[:2]
```
Comment (sethvargo) [inline internal/export/database/export_test.go:123]: Do we guarantee this won't panic? What if there is only 1 or no keys?

File: internal/export/database/export.go (hunk 1 above)
Comment (mikehelmick) [reply internal/export/database/export.go:321]: This is the best proxy we have as we don't capture create/effective times.

So relying on the serial column

File: internal/export/database/export_test.go (hunk 3 above)
Comment (mikehelmick) [reply internal/export/database/export_test.go:110]: changed around a bit, still creates an extra alloc... but we're talking about 2 pointers in a unit test

**Pull request merged**